language/tools/move-package/src/lib.rs

```diff
@@ -1,4 +1,49 @@
 // Copyright (c) The Diem Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
+pub mod compilation;
+pub mod resolution;
 pub mod source_package;
+
+use anyhow::Result;
+use compilation::compiled_package::CompiledPackage;
+use std::path::Path;
+use structopt::*;
+
+use crate::{
+    compilation::build_plan::BuildPlan,
+    resolution::resolution_graph::ResolutionGraph,
+    source_package::{layout, manifest_parser},
+};
+
+#[derive(Debug, StructOpt, Clone)]
+#[structopt(
+    name = "Move Package",
+    about = "Package and build system for Move code."
+)]
+pub struct BuildConfig {
+    /// Compile in 'dev' mode. The 'dev-addresses' and 'dev-dependencies' fields will be used
```

Comment:
You may want to compile package you're developing as a library but you can't do this since you have an uninstantiated named addresses you want to expose from it, e.g., `Std` in the MoveStdlib. In dev-mode you can:
* Instantiate any unassigned named addresses that are in scope in the `dev-addresses` section
* Include dev-only dependencies (think dev dependencies in cargo) for test-only code. 

These two things will only be done if you call `move package build -d` and eventually, dev dependencies will be included when running `move package test`.

Reply:
might want to add an explanation of why you would want these settings in the help for the tool :)